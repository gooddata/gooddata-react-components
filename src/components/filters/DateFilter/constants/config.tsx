// (C) 2007-2019 GoodData Corporation
import * as moment from "moment";
import { ExtendedDateFilters } from "@gooddata/typings";

import { platformDateFormat } from "../../../../constants/Platform";

export const defaultDateFilterOptions: ExtendedDateFilters.IDateFilterOptionsByType = {
    allTime: {
        localIdentifier: "ALL_TIME",
        type: "allTime",
        name: "",
        visible: true,
    },
    absoluteForm: {
        localIdentifier: "ABSOLUTE_FORM",
        type: "absoluteForm",
        from: moment()
            .subtract(1, "month")
            .startOf("day")
            .format(platformDateFormat),
        to: moment()
            .startOf("day")
            .format(platformDateFormat),
        name: "",
        visible: true,
    },
    absolutePreset: [],
    relativeForm: {
        localIdentifier: "RELATIVE_FORM",
        type: "relativeForm",
        granularity: "GDC.time.month",
        from: undefined,
        to: undefined,
        name: "",
        visible: true,
        availableGranularities: ["GDC.time.date", "GDC.time.month", "GDC.time.quarter", "GDC.time.year"],
    },
    relativePreset: {
        "GDC.time.date": [
            {
                from: -6,
                to: 0,
                granularity: "GDC.time.date",
                localIdentifier: "LAST_7_DAYS",
                type: "relativePreset",
                visible: true,
                name: "",
            },
            {
                from: -29,
                to: 0,
                granularity: "GDC.time.date",
                localIdentifier: "LAST_30_DAYS",
                type: "relativePreset",
                visible: true,
                name: "",
            },
            {
                from: -89,
                to: 0,
                granularity: "GDC.time.date",
                localIdentifier: "LAST_90_DAYS",
                type: "relativePreset",
                visible: true,
                name: "",
            },
        ],
        "GDC.time.month": [
            {
                from: 0,
                to: 0,
                granularity: "GDC.time.month",
                localIdentifier: "THIS_MONTH",
                type: "relativePreset",
                visible: true,
                name: "",
            },
            {
                from: -1,
                to: -1,
                granularity: "GDC.time.month",
                localIdentifier: "LAST_MONTH",
                type: "relativePreset",
                visible: true,
                name: "",
            },
            {
                from: -11,
                to: 0,
                granularity: "GDC.time.month",
                localIdentifier: "LAST_12_MONTHS",
                type: "relativePreset",
                visible: true,
                name: "",
            },
        ],
        "GDC.time.quarter": [
            {
                from: 0,
                to: 0,
                granularity: "GDC.time.quarter",
                localIdentifier: "THIS_QUARTER",
                type: "relativePreset",
                visible: true,
                name: "",
            },
            {
                from: -1,
                to: -1,
                granularity: "GDC.time.quarter",
                localIdentifier: "LAST_QUARTER",
                type: "relativePreset",
                visible: true,
                name: "",
            },
            {
                from: -3,
                to: 0,
                granularity: "GDC.time.quarter",
                localIdentifier: "LAST_4_QUARTERS",
                type: "relativePreset",
                visible: true,
                name: "",
            },
        ],
        "GDC.time.year": [
            {
                from: 0,
                to: 0,
                granularity: "GDC.time.year",
                localIdentifier: "THIS_YEAR",
                type: "relativePreset",
                visible: true,
                name: "",
            },
            {
                from: -1,
                to: -1,
                granularity: "GDC.time.year",
                localIdentifier: "LAST_YEAR",
                type: "relativePreset",
                visible: true,
                name: "",
            },
        ],
    },
};

export const defaultDateFilterProjectConfig: ExtendedDateFilters.IDateFilterConfigContent = {
    selectedOption: "THIS_MONTH",
    allTime: defaultDateFilterOptions.allTime,
    absoluteForm: defaultDateFilterOptions.absoluteForm,
    relativeForm: {
        // month has to be the first as it should be the default selected option
        granularities: ["GDC.time.month", "GDC.time.date", "GDC.time.quarter", "GDC.time.year"],
        localIdentifier: defaultDateFilterOptions.relativeForm.localIdentifier,
        name: defaultDateFilterOptions.relativeForm.name,
        visible: true,
    },
    relativePresets: Object.keys(defaultDateFilterOptions.relativePreset).reduce(
        (presets: ExtendedDateFilters.IDateFilterRelativePreset[], granularityKey: string) => {
            const granularityPresets: ExtendedDateFilters.IDateFilterRelativePreset[] =
                defaultDateFilterOptions.relativePreset[granularityKey];
            return [...presets, ...granularityPresets];
        },
        [],
    ),
    absolutePresets: [],
};
