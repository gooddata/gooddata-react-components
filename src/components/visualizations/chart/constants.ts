// (C) 2007-2018 GoodData Corporation
export const VIEW_BY_DIMENSION_INDEX = 1;
export const STACK_BY_DIMENSION_INDEX = 0;
export const PIE_CHART_LIMIT = 20;
export const HEATMAP_DATA_POINTS_LIMIT = 10000;

// for 'view by 2 attributes' or 'grouped attributes'
export const VIEW_BY_ATTRIBUTES_LIMIT = 2;
export const PARENT_ATTRIBUTE_INDEX = 0; // first attribute in 'View By' panel
export const PRIMARY_ATTRIBUTE_INDEX = 1; // second attribute in 'View By' panel
