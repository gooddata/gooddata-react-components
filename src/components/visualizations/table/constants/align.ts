// (C) 2007-2018 GoodData Corporation
export const ALIGN_LEFT = 'left';
export const ALIGN_RIGHT = 'right';
