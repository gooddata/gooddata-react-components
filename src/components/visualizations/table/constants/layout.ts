// (C) 2019 GoodData Corporation
export const DEFAULT_HEADER_HEIGHT: number = 26;
export const DEFAULT_ROW_HEIGHT: number = 30;
export const DEFAULT_FOOTER_ROW_HEIGHT: number = 30;

export const TOTALS_ADD_ROW_HEIGHT: number = 50;
export const TOTALS_TYPES_DROPDOWN_WIDTH: number = 150;
