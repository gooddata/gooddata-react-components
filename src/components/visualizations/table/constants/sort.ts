// (C) 2007-2018 GoodData Corporation
export const ASC = 'asc';
export const DESC = 'desc';
