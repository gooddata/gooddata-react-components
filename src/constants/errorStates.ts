// (C) 2007-2020 GoodData Corporation
export enum ErrorCodes {
    APP_NO_DATA = 1001,
    EMPTY_AFM = 1002,

    INVALID_BUCKETS = 2001,
}

export type CANCELLED = "CANCELLED";

export const ErrorStates = {
    /**
     * This error means that server could not understand the request due to invalid syntax.
     */
    BAD_REQUEST: "BAD_REQUEST",

    /**
     * This error means that location bucket is missing
     */
    GEO_LOCATION_MISSING: "GEO_LOCATION_MISSING",

    /**
     * This error means that you are not authorized.
     */
    UNAUTHORIZED: "UNAUTHORIZED",

    /**
     * This error means that executed data were too large to be displayed by GoodData.UI.
     */
    DATA_TOO_LARGE_TO_DISPLAY: "DATA_TOO_LARGE_TO_DISPLAY",

    /**
     * This error means that processed request would generate a result too large to be processed
     * by GoodData platform.
     */
    DATA_TOO_LARGE_TO_COMPUTE: "DATA_TOO_LARGE_TO_COMPUTE",

    /**
     * This error means that processed result contains negative values which does not make
     * sense within the given visualization (e.g. pie chart with negative values).
     */
    NEGATIVE_VALUES: "NEGATIVE_VALUES",

    /**
     * This error means that the processed result does not contain any data.
     */
    NO_DATA: "NO_DATA",

    /**
     * This error means that requested entity (e.g. a visualization) was not found on the server.
     */
    NOT_FOUND: "NOT_FOUND",

    /**
     * This error means that empty AFM was went to the GoodData.UI and as such can't be executed.
     */
    EMPTY_AFM: "EMPTY_AFM",

    /**
     * @internal
     */
    INVALID_BUCKETS: "INVALID_BUCKETS",

    /**
     * This error means that requested visualization is restricted by access rules within the GoodData platform.
     * Please contact your administrator.
     */
    PROTECTED_REPORT: "PROTECTED_REPORT",

    /**
     * This error means that GoodData.UI does not know how to handle such error.
     */
    UNKNOWN_ERROR: "UNKNOWN_ERROR",

    /**
     * This error means that request has been cancelled usually after component has been unmounted.
     */
    CANCELLED: "CANCELLED",
};
