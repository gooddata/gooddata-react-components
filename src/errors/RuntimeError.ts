// (C) 2007-2018 GoodData Corporation
export class RuntimeError extends Error {
    constructor(public message: string, public cause?: Error) {
        super(message);

        Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain
    }

    public getMessage() {
        return this.message;
    }

    public getCause() {
        return this.cause;
    }
}
