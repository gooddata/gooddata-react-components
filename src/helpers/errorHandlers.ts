// (C) 2007-2020 GoodData Corporation
import { Execution, VisualizationObject } from "@gooddata/typings";
import { ApiResponseError } from "@gooddata/gooddata-js";
import { IntlShape } from "react-intl";
import { ErrorStates, ErrorCodes } from "../constants/errorStates";
import get = require("lodash/get");
import includes = require("lodash/includes");
import * as HttpStatusCodes from "http-status-codes";
import { RuntimeError } from "../errors/RuntimeError";
import { unwrap } from "./utils";

function getJSONFromText(data: string): object {
    try {
        return JSON.parse(data);
    } catch (e) {
        return null;
    }
}

export function isApiResponseError(error: TypeError | ApiResponseError): error is ApiResponseError {
    return error && (error as ApiResponseError).response !== undefined;
}

export interface IErrorMap {
    [key: string]: {
        icon?: string;
        message: string;
        description: string;
    };
}

export function generateErrorMap(intl: IntlShape): IErrorMap {
    const errorMap = {
        [ErrorStates.DATA_TOO_LARGE_TO_DISPLAY]: {
            icon: "icon-cloud-rain",
            message: intl.formatMessage({ id: "visualization.ErrorMessageDataTooLarge" }),
            description: intl.formatMessage({ id: "visualization.ErrorDescriptionDataTooLarge" }),
        },
        [ErrorStates.GEO_LOCATION_MISSING]: {
            message: intl.formatMessage({ id: "geo.ErrorMessageLocationMissing" }),
            description: intl.formatMessage({ id: "geo.ErrorDescriptionLocationMissing" }),
        },
        [ErrorStates.NOT_FOUND]: {
            message: intl.formatMessage({ id: "visualization.ErrorMessageNotFound" }),
            description: intl.formatMessage({ id: "visualization.ErrorDescriptionNotFound" }),
        },
        [ErrorStates.UNAUTHORIZED]: {
            message: intl.formatMessage({ id: "visualization.ErrorMessageUnauthorized" }),
            description: intl.formatMessage({ id: "visualization.ErrorDescriptionUnauthorized" }),
        },
        [ErrorStates.NO_DATA]: {
            icon: "icon-filter",
            message: intl.formatMessage({ id: "visualization.ErrorMessageNoData" }),
            description: intl.formatMessage({ id: "visualization.ErrorDescriptionNoData" }),
        },
        [ErrorStates.UNKNOWN_ERROR]: {
            message: intl.formatMessage({ id: "visualization.ErrorMessageGeneric" }),
            description: intl.formatMessage({ id: "visualization.ErrorDescriptionGeneric" }),
        },
    };
    return errorMap;
}

// CAREFUL: error can also be of different type than listed
export function convertErrors(error: ApiResponseError | TypeError): RuntimeError {
    if (!isApiResponseError(error)) {
        return new RuntimeError(error.message, error);
    }

    const errorCode: number = error.response.status;
    switch (errorCode) {
        case HttpStatusCodes.NO_CONTENT:
            return new RuntimeError(ErrorStates.NO_DATA, error);

        case HttpStatusCodes.REQUEST_TOO_LONG:
            return new RuntimeError(ErrorStates.DATA_TOO_LARGE_TO_COMPUTE, error);

        case HttpStatusCodes.BAD_REQUEST:
            const message = get(getJSONFromText(error.responseBody), "error.message", "");

            if (includes(message, "Attempt to execute protected report unsafely")) {
                return new RuntimeError(ErrorStates.PROTECTED_REPORT, error);
            } else {
                return new RuntimeError(ErrorStates.BAD_REQUEST, error);
            }

        case HttpStatusCodes.NOT_FOUND:
            return new RuntimeError(ErrorStates.NOT_FOUND, error);

        case HttpStatusCodes.UNAUTHORIZED:
            return new RuntimeError(ErrorStates.UNAUTHORIZED, error);

        case ErrorCodes.EMPTY_AFM:
            return new RuntimeError(ErrorStates.EMPTY_AFM);

        case ErrorCodes.INVALID_BUCKETS:
            return new RuntimeError(ErrorStates.INVALID_BUCKETS);

        default:
            return new RuntimeError(ErrorStates.UNKNOWN_ERROR);
    }
}

/** @deprecated */
function isNullExecutionResult(responses: Execution.IExecutionResponses): boolean {
    return responses.executionResult === null;
}

function hasEmptyData(responses: Execution.IExecutionResponses): boolean {
    return responses.executionResult.data.length === 0;
}

function hasMissingHeaderItems(responses: Execution.IExecutionResponses): boolean {
    return !responses.executionResult.headerItems;
}

function isEmptyDataResult(responses: Execution.IExecutionResponses): boolean {
    return hasEmptyData(responses) && hasMissingHeaderItems(responses);
}

/**
 * isEmptyResult
 * is a function that returns true if the execution result is empty (no data points) and false otherwise.
 * @param responses:Execution.IExecutionResponses - object with execution response and result
 * @return boolean
 * @internal
 */
export function isEmptyResult(responses: Execution.IExecutionResponses): boolean {
    return isNullExecutionResult(responses) || isEmptyDataResult(responses);
}

export function checkEmptyResult(responses: Execution.IExecutionResponses) {
    if (isEmptyResult(responses)) {
        throw {
            name: "EmptyResultError",
            response: {
                status: HttpStatusCodes.NO_CONTENT,
                json: () => Promise.resolve(responses),
                text: () => Promise.resolve(null),
            },
        };
    }

    return responses;
}

export const hasDuplicateIdentifiers = (buckets: VisualizationObject.IBucket[]) => {
    const buffer: string[] = [];
    const duplicates: string[] = [];
    buckets.forEach(({ items }) => {
        items.forEach((bucketItem: VisualizationObject.BucketItem) => {
            const localIdentifier: string = unwrap(bucketItem).localIdentifier;
            const isDuplicate = buffer.includes(localIdentifier);
            buffer.push(localIdentifier);
            if (isDuplicate && !duplicates.includes(localIdentifier)) {
                duplicates.push(localIdentifier);
            }
        });
    });
    if (duplicates.length > 0) {
        // tslint:disable-next-line:no-console max-line-length
        console.warn(
            `Duplicate identifier${duplicates.length > 1 ? "s" : ""} '${duplicates.join(
                ", ",
            )}' detected in PivotTable. Please make sure all localIdentifiers are unique.`,
        );
    }
    return duplicates.length > 0;
};
