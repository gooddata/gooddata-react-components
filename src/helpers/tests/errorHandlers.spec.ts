// (C) 2007-2018 GoodData Corporation
import * as HttpStatusCodes from "http-status-codes";
import { checkEmptyResult, convertErrors, generateErrorMap, hasDuplicateIdentifiers } from "../errorHandlers";
import { ApiResponseError } from "@gooddata/gooddata-js";
import "isomorphic-fetch";

import {
    emptyResponse,
    emptyResponseWithNull,
    attributeOnlyResponse,
} from "../../execution/fixtures/ExecuteAfm.fixtures";
import { ErrorCodes, ErrorStates } from "../../constants/errorStates";
import { RuntimeError } from "../../errors/RuntimeError";
import "jest";
import { VisualizationObject } from "@gooddata/typings/dist";

async function createMockedError(status: number, body: string = "{}") {
    const response = new Response(body, { status });

    // In gooddata-js, the response body is always read before the rejectio with ApiResponseError,
    // see https://github.com/gooddata/gooddata-js/blob/c5c985e9070d20ac359b988244b7bb1155661473/src/xhr.ts#L154-L155
    const responseBody = await response.text();

    return new ApiResponseError("Response error", response, responseBody);
}

async function createTypeError() {
    return new TypeError("TypeError message");
}

describe("convertErrors", () => {
    it("should return RuntimeError with message when error type is not ApiResponseError", async () => {
        const e = convertErrors(await createTypeError());

        expect(e).toBeInstanceOf(RuntimeError);
        expect(e.message).toEqual("TypeError message");
    });

    it("should return `NO_DATA` error", async () => {
        const e = convertErrors(await createMockedError(204));

        expect(e).toBeInstanceOf(RuntimeError);
        expect(e.message).toEqual(ErrorStates.NO_DATA);
    });

    it("should return `DATA_TOO_LARGE_TO_COMPUTE` error", async () => {
        const e = convertErrors(await createMockedError(HttpStatusCodes.REQUEST_TOO_LONG));

        expect(e).toBeInstanceOf(RuntimeError);
        expect(e.message).toEqual(ErrorStates.DATA_TOO_LARGE_TO_COMPUTE);
    });

    it("should return `BAD_REQUEST` error", async () => {
        const e = convertErrors(await createMockedError(HttpStatusCodes.BAD_REQUEST));

        expect(e).toBeInstanceOf(RuntimeError);
        expect(e.message).toEqual(ErrorStates.BAD_REQUEST);
    });

    it("should return `UNAUTHORIZED` error", async () => {
        const e = convertErrors(await createMockedError(HttpStatusCodes.UNAUTHORIZED));

        expect(e).toBeInstanceOf(RuntimeError);
        expect(e.message).toEqual(ErrorStates.UNAUTHORIZED);
    });

    it("should return `NOT_FOUND` error", async () => {
        const e = convertErrors(await createMockedError(HttpStatusCodes.NOT_FOUND));

        expect(e).toBeInstanceOf(RuntimeError);
        expect(e.message).toEqual(ErrorStates.NOT_FOUND);
    });

    it("should return `PROTECTED_REPORT` error", async () => {
        const protectedErrorBody = `{
                "error": {
                    "message": "Attempt to execute protected report unsafely"
                }
            }`;

        const e = convertErrors(await createMockedError(HttpStatusCodes.BAD_REQUEST, protectedErrorBody));

        expect(e).toBeInstanceOf(RuntimeError);
        expect(e.message).toEqual(ErrorStates.PROTECTED_REPORT);
    });

    it("should return `EMPTY_AFM` error", async () => {
        const e = convertErrors(await createMockedError(ErrorCodes.EMPTY_AFM));

        expect(e).toBeInstanceOf(RuntimeError);
        expect(e.message).toEqual(ErrorStates.EMPTY_AFM);
    });

    it("should return `INVALID_BUCKETS` error", async () => {
        const e = convertErrors(await createMockedError(ErrorCodes.INVALID_BUCKETS));

        expect(e).toBeInstanceOf(RuntimeError);
        expect(e.message).toEqual(ErrorStates.INVALID_BUCKETS);
    });

    it("should return `UNKNOWN_ERROR` error", async () => {
        const e = convertErrors(await createMockedError(0));

        expect(e).toBeInstanceOf(RuntimeError);
        expect(e.message).toEqual(ErrorStates.UNKNOWN_ERROR);
    });
});

describe("checkEmptyResult", () => {
    it("should throw 204 if executionResult does not contain any data", () => {
        expect.hasAssertions();

        try {
            checkEmptyResult(emptyResponse);
        } catch (obj) {
            expect(obj.response.status).toEqual(204);
        }
    });

    it("should throw 204 if executionResult is null", () => {
        expect.hasAssertions();

        try {
            checkEmptyResult(emptyResponseWithNull);
        } catch (obj) {
            expect(obj.response.status).toEqual(204);
        }
    });

    it("should not throw 204 if executionResult does not contain any data, but contain headers", () => {
        expect(() => checkEmptyResult(attributeOnlyResponse)).not.toThrow();
    });
});

describe("generateErrorMap", () => {
    it("should generate map", () => {
        const intlMock = {
            formatMessage: ({ id }: { id: string }) => id,
        };
        const map = generateErrorMap(intlMock as any);
        expect(map).toMatchSnapshot();
    });
});

describe("hasDuplicateIdentifiers", () => {
    const buckets: VisualizationObject.IBucket[] = [
        {
            localIdentifier: "measures",
            items: [
                {
                    measure: {
                        localIdentifier: "abc",
                        definition: {
                            measureDefinition: {
                                item: {
                                    identifier: "aaEGaXAEgB7U",
                                },
                            },
                        },
                    },
                },
                {
                    measure: {
                        localIdentifier: "def",
                        definition: {
                            measureDefinition: {
                                item: {
                                    identifier: "aabHeqImaK0d",
                                },
                            },
                        },
                    },
                },
            ],
        },
        {
            localIdentifier: "rows",
            items: [
                {
                    visualizationAttribute: {
                        localIdentifier: "ghi",
                        displayForm: {
                            identifier: "label.restaurantlocation.locationstate",
                        },
                    },
                },
                {
                    visualizationAttribute: {
                        localIdentifier: "jkl",
                        displayForm: {
                            identifier: "label.restaurantlocation.locationname",
                        },
                    },
                },
                {
                    visualizationAttribute: {
                        localIdentifier: "abc",
                        displayForm: {
                            identifier: "label.menuitem.menucategory",
                        },
                    },
                },
            ],
        },
        {
            localIdentifier: "columns",
            items: [
                {
                    visualizationAttribute: {
                        localIdentifier: "def",
                        displayForm: {
                            identifier: "date.aam81lMifn6q",
                        },
                    },
                },
                {
                    visualizationAttribute: {
                        localIdentifier: "xyz",
                        displayForm: {
                            identifier: "date.abm81lMifn6q",
                        },
                    },
                },
            ],
        },
    ];
    it("should return true if there are duplicate identifiers", () => {
        const hasDuplicates = hasDuplicateIdentifiers(buckets);
        expect(hasDuplicates).toBe(true);
    });
    it("should return false if there are duplicate identifiers", () => {
        const hasDuplicates = hasDuplicateIdentifiers(buckets.slice(1));
        expect(hasDuplicates).toBe(false);
    });
});
