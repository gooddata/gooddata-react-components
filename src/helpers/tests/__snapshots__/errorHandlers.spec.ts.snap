// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateErrorMap should generate map 1`] = `
Object {
  "DATA_TOO_LARGE_TO_DISPLAY": Object {
    "description": "visualization.ErrorDescriptionDataTooLarge",
    "icon": "icon-cloud-rain",
    "message": "visualization.ErrorMessageDataTooLarge",
  },
  "NOT_FOUND": Object {
    "description": "visualization.ErrorDescriptionNotFound",
    "message": "visualization.ErrorMessageNotFound",
  },
  "NO_DATA": Object {
    "description": "visualization.ErrorDescriptionNoData",
    "icon": "icon-filter",
    "message": "visualization.ErrorMessageNoData",
  },
  "UNAUTHORIZED": Object {
    "description": "visualization.ErrorDescriptionUnauthorized",
    "message": "visualization.ErrorMessageUnauthorized",
  },
  "UNKNOWN_ERROR": Object {
    "description": "visualization.ErrorDescriptionGeneric",
    "message": "visualization.ErrorMessageGeneric",
  },
}
`;
